### PR TITLE
Remove phase name from moped_proj_phases

### PR DESCRIPTION
**Pull request by chiaberry**
## Associated issues

https://github.com/cityofaustin/atd-data-tech/issues/9350

This is the second part of normalizing the moped_phases and moped_proj_phases tables. In this PR I have updated the project_list_view to not rely on the phase_name from moped_proj_phases and updated the Signal Projects table query as well. 

I also ripped the band aid and dropped the phase_name column. Ive gone through multiple times looking for references to the phase_name column on the moped_proj_phases table in the editor and think I removed them all, but would really appreciate some kicking of the tires. 

## Testing
**URL to test:** 
https://9350-normal-phase--atd-moped-main.netlify.app/moped/projects

**Steps to test:**

Add a construction phase (with start date) or completion phase (with end date) to a project. 
Check that the dates appear in the projects list. 

Add a project with a signal, add a phase and mark as current. Go to the signals view and see the name of the current phase. 
https://9350-normal-phase--atd-moped-main.netlify.app/moped/views/signal-projects

_There should not be any change in functionality from existing behavior in moped._ 

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/queries/project.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const ADD_PROJECT = gql`
       fiscal_year
       capitally_funded
       moped_proj_phases {
-        phase_name
+        phase_id
```
Comment (chiaberry) [inline moped-editor/src/queries/project.js:16]: In this PR I learned that even if you don't explicitly list the field in the insert mutation, information included in the object being sent will still make its way into the database insertion

File: moped-database/views/project_list_view.sql [hunk 2]
```diff
@@ -81,7 +81,7 @@ AS WITH project_person_list_lookup AS (
       FROM moped_proj_phases phases
       WHERE true
         AND phases.project_id = mp.project_id 
-        AND phases.phase_name = 'construction'::text
```
Comment (johnclary) [inline moped-database/views/project_list_view.sql:84]: i wonder if we might want to leave this as-is (matching on the `phase_name` text). it feels fragile to rely on the ID sequence.

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 3]
```diff
@@ -532,9 +532,6 @@ const ProjectTimeline = ({ refetch: refetchSummary }) => {
                         project_id: projectId,
                         completion_percentage: 0,
                         completed: false,
-                        // temporary until project phase normalization is complete
-                        phase_name:
-                          phaseNameLookup[newData?.phase_id].toLowerCase(),
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:537]: 🚀

File: moped-editor/src/views/projects/projectView/ProjectTimeline.js [hunk 4]
```diff
@@ -597,12 +594,6 @@ const ProjectTimeline = ({ refetch: refetchSummary }) => {
                         ["phase_id", "is_current_phase"].includes(value)
                       ).length > 0;
 
-                    // temporary workaround until phase normalization is complete
-                    if (currentPhaseChanged) {
-                      updatedPhaseObject["phase_name"] =
-                        phaseNameLookup[newData.phase_id].toLowerCase();
-                    }
-
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectTimeline.js:605]: 🚀

File: moped-database/views/project_list_view.sql (hunk 2 above)
Comment (johnclary) [reply moped-database/views/project_list_view.sql:84]: edit: ignore me! phase_name is not available, and i don't think we need to take the extra step of filtering on the `moped_phases.phase_name`

File: moped-database/migrations/1658436152142_use_phase_ids_instead_of_names/up.sql [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        AND phases.phase_id = 9 -- phase_id 9 is construction
+        AND phases.is_deleted = false
+      ORDER BY phases.date_added DESC
+      LIMIT 1) AS construction_start_date,
+    ( -- get me the phase end of the most recently added completion phase entry
+      SELECT phases.phase_end
+      FROM moped_proj_phases phases
+      WHERE true 
+        AND phases.project_id = mp.project_id 
+        AND phases.phase_id = 11 -- phase_id 11 is complete
```
Comment (frankhereford) [inline moped-database/migrations/1658436152142_use_phase_ids_instead_of_names/up.sql:93]: I did confirm that these match the view in place on the test deployment.

**Review by frankhereford — CHANGES_REQUESTED** [review]
When I peeked this on its test deployment (https://9350-normal-phase--atd-moped-main.netlify.app/moped), I didn't see the start and end date displayed on the project listing page for the construction and completion phases I had added. I did make sure that the migrations and run too .. ✅

The current phase name did show up in the badge as expected.

Let me know if I'm missing something, I'll be happy to flip this to an approval if I'm out in left field.

**Review by frankhereford — APPROVED** [review]
mybad

Woops - my error! By testing it on test without the contents of [this PR #746](https://github.com/cityofaustin/atd-moped/pull/746), the test netlify deployment was connecting to staging's graphql endpoint.

This is A-OK - thank you @chiaberry!

**Comment by frankhereford** [comment]
For the curious, this is deployed to test: https://9350-normal-phase--atd-moped-main.netlify.app/moped/projects

**Review by patrickm02L — APPROVED** [review]
Everything checks out! 🤘

https://user-images.githubusercontent.com/5566727/184182359-9a76697c-c08c-40d9-ae83-0a8a4535e53e.mp4

�

**Review by mateoclarke — COMMENTED** [review]
👍

**Pull request merged**